Source/FieldSolver/SpectralSolver/SpectralSolverRZ.H

```diff
@@ -0,0 +1,104 @@
+                               int const field_index,
+                               int const i_comp=0 ) {
+            BL_PROFILE("SpectralSolverRZ::ForwardTransform");
+            field_data.ForwardTransform(field_mf, field_index, i_comp);
+        };
+
+        /* \brief Transform the two MultiFabs `field_mf1` and `field_mf2`
+         *  to spectral space, and store the corresponding results internally
+         *  (in the spectral field specified by `field_index1` and `field_index2`) */
+        void ForwardTransform (amrex::MultiFab const & field_mf1, int const field_index1,
+                               amrex::MultiFab const & field_mf2, int const field_index2) {
+            BL_PROFILE("SpectralSolverRZ::ForwardTransform");
+            field_data.ForwardTransform(field_mf1, field_index1,
+                                        field_mf2, field_index2);
+        };
+
+        /* \brief Transform spectral field specified by `field_index` back to
+         * real space, and store it in the component `i_comp` of `field_mf` */
+        void BackwardTransform (amrex::MultiFab& field_mf,
+                                int const field_index,
+                                int const i_comp=0) {
+            BL_PROFILE("SpectralSolverRZ::BackwardTransform");
+            field_data.BackwardTransform(field_mf, field_index, i_comp);
+        };
+
+        /* \brief Transform spectral fields specified by `field_index1` and `field_index2`
+         * back to real space, and store it in `field_mf1` and `field_mf2`*/
+        void BackwardTransform (amrex::MultiFab& field_mf1, int const field_index1,
+                                amrex::MultiFab& field_mf2, int const field_index2) {
+            BL_PROFILE("SpectralSolverRZ::BackwardTransform");
+            field_data.BackwardTransform(field_mf1, field_index1,
+                                         field_mf2, field_index2);
+        };
+
+        /* \brief Update the fields in spectral space, over one timestep */
+        void pushSpectralFields () {
+            BL_PROFILE("SpectralSolverRZ::pushSpectralFields");
+            // Virtual function: the actual function used here depends
+            // on the sub-class of `SpectralBaseAlgorithm` that was
+            // initialized in the constructor of `SpectralSolverRZ`
+            algorithm->pushSpectralFields(field_data);
+        };
+
+        /**
+          * \brief Public interface to call the member function ComputeSpectralDivE
+          * of the base class SpectralBaseAlgorithmRZ from objects of class SpectralSolverRZ
+          */
+        void ComputeSpectralDivE ( const std::array<std::unique_ptr<amrex::MultiFab>,3>& Efield,
+                                   amrex::MultiFab& divE ) {
+            algorithm->ComputeSpectralDivE( field_data, Efield, divE );
+        };
+
+    private:
+
+        SpectralHankelKSpace k_space; // Save the instance to initialize filtering
```

Comment:
I am being a bit picky here, but I would suggest to remove this line and add it only in the PR on filtering.
```suggestion
```

Reply:
Yes, I can revert and move to the k-space filtering branch.